### PR TITLE
vtm-gdx-poi3d module

### PR DESCRIPTION
**Pull request by schedul-xor**
image

Moved poi3d to vtm-gdx-models and imported from vtm-web-app.

### CODE REVIEW

**Comment by devemux86** [comment]
Thanks, see my comment [here](https://github.com/mapsforge/vtm/issues/580#issuecomment-436536028).

**Comment by devemux86** [comment]
@gustl22 any more work on 3D classes or can be moved and where?

File: vtm-gdx-models/src/VtmGdxModels.gwt.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?><!DOCTYPE module PUBLIC
+        "-//Google Inc.//DTD Google Web Toolkit trunk//EN"
+        "http://google-web-toolkit.googlecode.com/svn/trunk/distro-source/core/src/gwt-module.dtd">
+<module rename-to="VtmGdxModels">
+    <source path="org/oscim/model/poi3d" />
+</module>
```
[the remark this answers is not in the document]
Comment (schedul-xor) [reply vtm-gdx-models/src/VtmGdxModels.gwt.xml]: Tried moving <source> tags to vtm-web/src/org/oscim/gdx/VtmWeb.gwt.xml, but it didn't work.

    Compiling module org.oscim.web.VtmWebApp
       Finding entry point classes
          Tracing compile failure path for type 'org.oscim.web.client.GwtLauncher'
             [ERROR] Errors in 'file:/home/xor/local/m/p/w/vtm/vtm-web-app/src/org/oscim/web/client/GwtLauncher.java'
                [ERROR] Line 47: No source code is available for type org.oscim.gdx.client.MapConfig; did you forget to inherit a required module?
                [ERROR] Line 47: No source code is available for type org.oscim.core.Tile; did you forget to inherit a required module?
             [ERROR] Errors in 'com/badlogic/gdx/backends/gwt/GwtApplicationConfiguration.java'
                [ERROR] com.badlogic.gdx.backends.gwt.GwtGraphics.OrientationLockType cannot be resolved to a type
             [ERROR] Errors in 'file:/home/xor/local/m/p/w/vtm/vtm-web/src/org/oscim/gdx/emu/com/badlogic/gdx/backends/gwt/GwtApplication.java'
                [ERROR] com.badlogic.gdx.backends.gwt.GwtGraphics cannot be resolved to a type
             [ERROR] Errors in 'file:/home/xor/local/m/p/w/vtm/vtm-web/src/org/oscim/gdx/emu/com/badlogic/gdx/backends/gwt/GwtGraphics.java'
                [ERROR] Line 117: No source code is available for type org.oscim.gdx.client.GdxGL; did you forget to inherit a required module?

Source paths are relative.

**Comment by Gustl22** [comment]
> @Gustl22 any more work on 3D classes or can be moved and where?

I wanted to solve #601 before make it available to other platforms. But I didn't found the bug yet.
Wanted to move it to `vtm-gdx/src/org/oscim/gdx/poi3d`. I think package name `model` is slightly too abstract. Maybe you find a better name, but this is not that important for me.
Second is that `vtm-gdx` would need the `vtm-model` dependencies, which makes gdx package heavier (?)

**Comment by devemux86** [comment]
> Wanted to move it to vtm-gdx/src/org/oscim/gdx/poi3d. I think package name model is slightly too abstract. Maybe you find a better name, but this is not that important for me. model is ok, too.

Most suitable package seems the `org.oscim.gdx.poi3d` ("similar" to existing one).

> Second is that vtm-gdx would need the vtm-model dependencies, which makes gdx package heavier (?)

In that case we can create a new `vtm-poi3d` module to host the classes.

**Comment by Gustl22** [comment]
Agreed, if @shedul-xor doesn't mind :)

**Comment by schedul-xor** [comment]
No problem!

**Comment by Gustl22** [comment]
Now that #601 should be solved with #604, @devemux86 is invited to merge your PR (eventually splitted in one refactoring part and one PR for web implementation only). Afterwards I will add support for Desktop and Android.

**Comment by devemux86** [comment]
Thanks!
Merged via 09a6cf2497c0caffb21ccc747ef03f93a0ebb30b.

**Pull request closed without merging**